pennylane/ftqc/utils.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains utility data-structures and algorithms supporting functionality in the
+ftqc module.
+"""
+from threading import RLock
+
+
+class QubitMgr:
+    """
+    The QubitMgr object maintains a list of active and inactive qubit wire indices used and for use
+    during execution of a workload. Its purpose is to allow tracking of free qubit indices that
+    are in the |0> state to participate in MCM-based workloads, under the assumption of reset
+    upon measurement. Qubit wires indices will be tracked with a monotonically increasing set
+    of values, starting from the initial input `start_idx`.
+
+    This class assumes single-producer, single-consumer serialized CRUD operations only, and may
+    not behave correctly in a concurrent execution environment.
```

Comment:
Is it still the case after adding the `rlock`?

Reply:
I'll update this :+1:

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7075/commits/1e2accf03143e6ae1f2955827d21ca124d80ab47